### PR TITLE
expression: workaround for wrong timediff function input

### PR DESCRIPTION
**Pull request by tiancaiamao**
there is bug that builtinTimeDiffSig.eval's input is not legal:
row[0] row[1] may be a datetime with Fsp=-1

this COMMIT is a workaroud for the case

-----------------------------------

Input SQL:

```
CREATE TABLE `t` (
  `create_at` datetime NOT NULL DEFAULT '1000-01-01 00:00:00',
  `finish_at` datetime NOT NULL DEFAULT '1000-01-01 00:00:00'
);

INSERT INTO t values ('2016-02-13 15:32:24',  '2016-02-11 17:23:22');
SELECT timediff(finish_at, create_at) from t;
```

The session would panic:

```
2017/02/08 16:09:20 conn.go:330: [error] lastCmd select timediff(finish_at, create_at) from t, runtime error: slice bounds out of range, goroutine 98 [running]:
github.com/pingcap/tidb/server.(*clientConn).Run.func1(0xc420204000)
	/media/genius/OS/project/src/github.com/pingcap/tidb/server/conn.go:329 +0xe2
panic(0xce1480, 0xc42000c160)
	/media/genius/OS/project/go/src/runtime/panic.go:458 +0x243
github.com/pingcap/tidb/util/types.Duration.formatFrac(0xffff68e50d1b9400, 0xffff, 0x0, 0x1, 0x0)
	/media/genius/OS/project/src/github.com/pingcap/tidb/util/types/time.go:656 +0xd9
github.com/pingcap/tidb/util/types.Duration.String(0xffff68e50d1b9400, 0xffff, 0x7f3174d92e10, 0xc4203b37f8)
	/media/genius/OS/project/src/github.com/pingcap/tidb/util/types/time.go:646 +0x36a
github.com/pingcap/tidb/server.dumpTextValue(0xc42020400b, 0xffff0009, 0xffff68e50d1b9400, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1510500, 0x10, ...)
	/media/genius/OS/project/src/github.com/pingcap/tidb/server/util.go:330 +0x57e
github.com/pingcap/tidb/server.(*clientConn).writeResultset(0xc420204000, 0x12eb440, 0xc420664b00, 0xc420660000, 0x0, 0x0)
	/media/genius/OS/project/src/github.com/pingcap/tidb/server/conn.go:705 +0x65f
github.com/pingcap/tidb/server.(*clientConn).handleQuery(0xc420204000, 0xc420663381, 0x2c, 0x0, 0x0)
	/media/genius/OS/project/src/github.com/pingcap/tidb/server/conn.go:600 +0x182
github.com/pingcap/tidb/server.(*clientConn).dispatch(0xc420204000, 0xc420663381, 0x2d, 0x2c, 0x0, 0x0)
	/media/genius/OS/project/src/github.com/pingcap/tidb/server/conn.go:390 +0x5f5
github.com/pingcap/tidb/server.(*clientConn).Run(0xc420204000)
	/media/genius/OS/project/src/github.com/pingcap/tidb/server/conn.go:345 +0x10a
github.com/pingcap/tidb/server.(*Server).onConn(0xc420bd40c0, 0x12f3120, 0xc420141fe0)
	/media/genius/OS/project/src/github.com/pingcap/tidb/server/server.go:209 +0x13e
created by github.com/pingcap/tidb/server.(*Server).Run
	/media/genius/OS/project/src/github.com/pingcap/tidb/server/server.go:173 +0x96

```

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL
we should a real **fix** instead of workaround, I'll create a issue, would you like to take it @hanfei1991

**Comment by hanfei1991** [comment]
any test ?

**Comment by hanfei1991** [comment]
LGTM

**Comment by hanfei1991** [comment]
@coocood @zimulala PTAL

File: expression/builtin_time.go [hunk 1]
```diff
@@ -255,6 +255,9 @@ func (b *builtinTimeDiffSig) eval(row []types.Datum) (d types.Datum, err error)
 	}
 
 	t := t1.Sub(&t2)
+	// TODO: It should be the larger between t1.Fsp and t2.Fsp, rather than types.MaxFsp
```
Comment (zimulala) [inline expression/builtin_time.go]: Add . at the end of a line.

**Comment by zimulala** [comment]
LGTM

**Pull request merged**